File: TMS-MobileApp/src/app/types/traject.ts
// types/traject.ts
export interface ITraject {
  id: number;
  name: string;
  createdAt: string;
  updatedAt?: string;
  points: ITrajectPoint[];
  startLocationId: number;
  endLocationId: number;
  isPredefined: boolean;

}

export interface ITrajectPoint {
  id?: number;
  location?: string;
  order: number;
  trajectId?: number;
  clientId?: number;
  clientName?: string;
}

export interface ICreateTrajectDto {
  name: string;
  points: ICreateTrajectPointDto[];
}

export interface ICreateTrajectPointDto {
  location: string;
  order: number;
  clientId?: number;
}

export interface IUpdateTrajectDto {
  name?: string;
  points?: ICreateTrajectPointDto[];
}
export interface IPagedTrajectData {
  data: ITraject[];
  totalData: number;
}

// Options pour le formulaire
export interface ITrajectFormData {
  trajectId?: number;
}

File: backend/TransportManagementSystem/Entity/Traject.cs
using System.ComponentModel.DataAnnotations;

namespace TransportManagementSystem.Entity;

public class Traject
{
    public int Id { get; set; }

    [Required]
    public string Name { get; set; }
    public List<TrajectPoint> Points { get; set; } = new List<TrajectPoint>();

    public int? StartLocationId { get; set; }
    public int? EndLocationId { get; set; }
    public bool IsPredefined { get; set; } = false;
    public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;

}